packages/ifpack2/src/Ifpack2_Details_Filu_def.hpp

```diff
@@ -0,0 +1,289 @@
+  computeTime = localPrec->getComputeTime();
+}
+
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+bool Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+isComputed() const
+{
+  return computedFlag;
+}
+
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+Teuchos::RCP<const Tpetra::RowMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> >
+Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getMatrix() const
+{
+  return mat;
+}
+
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+int Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getNumInitialize() const
+{
+  return nInit;
+}
+
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+int Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getNumCompute() const
+{
+  return nComputed;
+}
+
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+int Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getNumApply() const
+{
+  return nApply;
+}
+
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+double Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getInitializeTime() const 
+{
+  return initTime;
+}
+
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+double Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getComputeTime() const
+{
+  return computeTime;
+}
+
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+double Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getApplyTime() const
+{
+  return applyTime;
+}
+
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+void Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+checkLocalILU()
+{
+  localPrec->checkILU();
+}
+
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+void Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+checkLocalIC()
+{
+  localPrec->checkIC();
+}
+
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+void Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+printStatus()
+{
+  std::cout << "FAST-ILU Preconditioner" << std::endl;
+  std::cout << "ILU - Non-symmetric" << std::endl;
+  std::cout << "n = " << mat->getNodeNumRows() << std::endl;
+  std::cout << "nnz = " << mat->getNodeNumEntries() << std::endl;
+  std::cout << "sweeps = " << localPrec->getNFact() << std::endl;
+  std::cout << "t init = " << initTime << "s" << std::endl; 
+  std::cout << "t comp = " << computeTime << "s" << std::endl; 
+  std::cout << "t appl = " << applyTime << "s" << std::endl; 
+  localPrec->checkILU();
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+void Filu<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+setMatrix(const Teuchos::RCP<const TRowMatrix>& A)
+{
+  //bmk note: this modeled after RILUK::setMatrix
+  if(mat.get() != A.get())
+  {
+    mat = A;
+    computedFlag = false;
```

Comment:
See above note; need to unset initialized flag as well, unless graph pointers are the same.